src/App.tsx

```diff
@@ -1,24 +1,337 @@
-/* eslint-disable max-len */
+/* eslint-disable prefer-const */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { getTodos, createTodo, deleteTodo } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem';
+
+const USER_ID = 10613;
+
+enum FilterStatus {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
+
+enum ErrorType {
+  onLoad,
+  onAdd,
+  onDelete,
+  onUpdate,
+  missingTitle,
+}
+
+type SetError = {
+  type: 'set_error';
+  value: ErrorType;
+};
+
+type RemoveError = {
+  type: 'remove_error';
+};
+
+const reducer = (state: {
+  isError: boolean,
+  errorType: ErrorType | null,
+}, action: SetError | RemoveError) => {
+  switch (action.type) {
+    case 'set_error':
+      return {
+        errorType: action.value,
+        isError: true,
+      };
+
+    case 'remove_error':
+      return {
+        ...state,
+        isError: false,
+      };
+
+    default:
+      return state;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [processedTodoIds, setProcessedTodoIds] = useState<number[]>([]);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.all);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isCreating, setIsCreating] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, dispatch] = useReducer(reducer, {
```

Comment:
that's nice you figured out with `useReducer` - :+1: 
but here is a bit overcoding (now you even have everything in one component) - everything you need is state with `ErrorType` enum value (None by default)
and just set correct type of error when it's needed